libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3245,7 +3255,12 @@ IRVariable IRGeneratorForStatements::readFromLValue(IRLValue const& _lvalue)
 					")\n";
 			}
 			else
-				define(result) << "loadimmutable(\"" << std::to_string(_immutable.variable->id()) << "\")\n";
+			{
+				if (!m_context.eofVersion().has_value())
+					define(result) << "loadimmutable(\"" << std::to_string(_immutable.variable->id()) << "\")\n";
+				else
+					define(result) << "auxdataloadn(" << std::to_string(m_context.immutableMemoryOffset(*_immutable.variable, true)) << ")\n";
```

Comment:
Boolean flags are not very readable, so it's best to include the parameter name most of the time. Or use an enum. Or define a separate function with distinct name - that is probably the least annoying option when we have only one such parameter.
```suggestion
					define(result) << "auxdataloadn(" << std::to_string(m_context.immutableMemoryOffset(*_immutable.variable, true /* _relative */)) << ")\n";
```

Reply:
done